Document Barrier Reverse Convertible (#431)
digital-asset/docs.daml.com: docs/2.8.0/docs/daml-finance/instruments/structured-product.rst

.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
.. SPDX-License-Identifier: Apache-2.0

> How To Use the Structured Product Extension Package

Reviewer: In one of the team meetings during your holiday we discussed dropping the "Extension" nomenclature. We don't define anywhere in the docs what an extension package is, so I refactored the other pages to avoid using "extension".

```suggestion
How To Use the Structured Product Instrument Packages
```

I am also using the plural as we are effectively using two packages here (interface + implementation)

Author: Maybe we want to define Extension as the two packages here (interface + implementation). That would make it singular, which I think is less confusing (somebody might wonder what other packages we are referring to).

Reviewer: let's discuss in the catchup?

> How To Use the Structured Product Instrument Package

Reviewer: I we agreed that this should be singular, we should be consistent and do it across all instrument pages

####################################################

To follow the script used in this tutorial, you can
`clone the Daml Finance repository <https://github.com/digital-asset/daml-finance>`_. In particular,
the Structured Product test folder
`Instrument/StructuredProduct/Test/ <https://github.com/digital-asset/daml-finance/blob/main/src/test/daml/Daml/Finance/Instrument/StructuredProduct/Test/>`_
is the starting point of this tutorial.

> How To Create a Structured Product Instrument

Reviewer: I would align with the other pages, such as 

```suggestion
How to create a Structured Product Instrument
```

Author: Not sure what the correct standard is, so I opened this: https://github.com/digital-asset/daml-finance/issues/1004
(covers other aspects as well).

Reviewer: thanks 👍 shall we align the format anyway and then let them review?

Author: I think it is better to wait and do it right once.

*********************************************

In order to create a structured product instrument, you first have to decide what type of payoff you
need. The

> :doc:`structured product extension package <../packages/implementations/daml-finance-instrument-structuredproduct>`

Reviewer: ```suggestion
:doc:`structured product instrument packages <../packages/implementations/daml-finance-instrument-structuredproduct>`
```
